features/checkout/paying_for_order/preventing_to_change_payment_method_of_already_cancelled_order.feature

```diff
@@ -0,0 +1,31 @@
+@paying_for_order
+Feature: Preventing to change payment method of the cancelled order
+    In order to not be able to change payment method of the order that is already cancelled
+    As a Customer
+    I want to be prevented from changing payment method of the cancelled order
+
+    Background:
+        Given the store operates on a single channel in "United States"
+        And the store has a product "Angel T-Shirt"
+        And the store ships everywhere for Free
+        And the store allows paying with "Cash on Delivery"
+        And the store also allows paying with "Bank Transfer"
+        And I am a logged in customer
+        And I placed an order "#00000666"
+        And I bought a single "Angel T-Shirt"
+        And I addressed it to "Lucifer Morningstar", "Seaside Fwy", "90802" "Los Angeles" in the "United States" with identical billing address
+        And I chose "Free" shipping method with "Cash on Delivery" payment
+
+    @api @no-ui
+    Scenario: Being prevented to change payment method of cancelled order
+        Given this order was cancelled
+        When I change my payment method to "Bank Transfer"
+        Then I should be notified that I can no longer change payment method of this order
+
+    @ui @no-api
+    Scenario: Being prevented to change payment method of cancelled order
+        When I browse my orders
+        And I am on the change payment method page
+        Then this order was cancelled
+        And I switch to "Bank Transfer" payment method
+        And I should be notified that I can no longer change payment method of this order
```

Comment:
```suggestion
	@ui @api
	Scenario: Being prevented to change payment method of cancelled order
	Given this order was cancelled
	And I want to change this order's payment method
	When I try to change my payment method to "Bank Transfer"
	Then I should be notified that I can no longer change payment method of this order
```

I believe this scenario is good enough (it can be improved anyway 🖖🏻) and covers both UI and API at once.

Reply:
The steps differs to much to make it into one scenario, in my opinion it's really similar case to "thank you page". 
When it comes to UI i need to be on the change payment method's page before it is cancelled, when it comes to API i don't need to come accross this, I mean i can add empty step `I am on the change payment method page` to API, but I dont feel it necessary

Reply:
> And I want to change this order's payment method
in API it might be a facade for preparing an API request
in UI it might be a facade for browsing and checking whether I am on the right page

Reply:
In scenarios, we should think about how to describe features without thinking about the steps implementation,
sometimes we should describe the entire process step by step sometimes not. how to make a decision about which way is better? When we are describing the process we probably should describe all steps, when we are describing specific features, all not important steps should be hidden.
In this case, you describe validation, not the entire process of changing state, so Jakub's proposition looks nice.
step `And I want to change this order's payment method` in UI context should open changing payment page, and in API context should prepare the request.
 BTW @jakubtobiasz , `when` steps shouldn't check anything, so browsing the proper page will be enough :)

Reply:
Mea culpa, I meant "opening the right page", not checking :D